gnovm/pkg/gnolang/gnomod.go

```diff
@@ -0,0 +1,115 @@
+package gnolang
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/gnovm/pkg/packages"
+	"github.com/gnolang/gno/tm2/pkg/std"
+)
+
+const (
+	// gno.mod files assumed in testing/default contexts.
+	GnoModLatest  = `go 0.9` // when gno.mod is missing for stdlibs.
+	GnoModTesting = `go 0.9` // when gno.mod is missing while testing.
+	GnoModDefault = `go 0.0` // when gno.mod is missing in general.
+	GnoVerDefault = `0.0`    // when gno version isn't specified.
+)
+
+// ========================================
+// Parses and checks the gno.mod file from mpkg.
+// To generate default ones, use:
+// gnomod.ParseBytes(GnoModDefault)
+//
+// Results:
+//   - mod: the gno.mod file, or nil if not found.
+//   - err: wrapped error
+func ParseCheckGnoMod(mpkg *std.MemPackage) (mod *gnomod.File, err error) {
+	if IsStdlib(mpkg.Path) {
+		// stdlib/extern packages are assumed up to date.
+		mod, _ = gnomod.ParseBytes("<stdlibs>/gno.mod", []byte(GnoModLatest))
+	} else if mod, _ = gnomod.ParseMemPackage(mpkg); mod == nil {
+		// gno.mod doesn't exist.
+		mod, _ = gnomod.ParseBytes(mpkg.Path+"/gno.mod", []byte(GnoModDefault))
```

Comment:
All these things lead me to believe that we're accumulating technical debt because we will desire more than just gno.mod.
Voting to switch to TOML now.